// wxWindows Patch
// Modified to be compatible with current unreleased library
// 10-24-2001 Robert Cavanaugh
// Description:
// This patch enables the wxWizard class to use .WXR resource
// files to compose the individual pages. Currently the
// library attempts to load all the resource information
// into the first page, resulting in an unreadable mess

--- Tip/wxWindows/include/wx/wizard.h	Sat Sep 15 04:39:10 2001
+++ New/wxWindows/include/wx/wizard.h	Fri Oct 19 17:30:56 2001
@@ -5,5 +5,7 @@
 // Author:      Vadim Zeitlin (partly based on work by Ron Kuris and Kevin B.
 //              Smith)
-// Modified by:
+// Modified by:	Robert Cavanaugh
+//              Added capability to use .WXR resource files in Wizard pages
+//              Added wxWIZARD_HELP event
 // Created:     15.08.99
 // RCS-ID:      $Id: wizard.h,v 1.13 2001/09/15 10:39:10 VS Exp $
@@ -49,5 +51,9 @@
     // that no other parameters are needed because the wizard will resize and
     // reposition the page anyhow
-    wxWizardPage(wxWizard *parent, const wxBitmap& bitmap = wxNullBitmap);
+    // Modified 8/20/2001 Robert Cavanaugh
+	// Add an optional resource string to build the page controls
+    wxWizardPage(wxWizard *parent, 
+                const wxBitmap& bitmap = wxNullBitmap,
+                char* resource = NULL);
 
     // these functions are used by the wizard to show another page when the
@@ -60,8 +66,8 @@
     // dynamically or to do something even more fancy. It's ok to return
     // wxNullBitmap from here - the default one will be used then.
-    virtual wxBitmap GetBitmap() const { return m_bitmap; }
+    virtual wxBitmap GetBitmap() const { return m_PageBitmap; }
 
 protected:
-    wxBitmap m_bitmap;
+    wxBitmap m_PageBitmap;
 
 private:
@@ -82,8 +88,13 @@
 public:
     // ctor takes the previous and next pages
+    // Modified 8/20/2001 Robert Cavanaugh
+	// Add an optional resource string to build the page controls
+    // Since the resource is added, add back the bitmap parameter
     wxWizardPageSimple(wxWizard *parent = NULL, // let it be default ctor too
                        wxWizardPage *prev = (wxWizardPage *)NULL,
-                       wxWizardPage *next = (wxWizardPage *)NULL)
-        : wxWizardPage(parent)
+                       wxWizardPage *next = (wxWizardPage *)NULL,
+                       const wxBitmap& bitmap = wxNullBitmap,
+                       char* resource = NULL)
+        : wxWizardPage(parent, bitmap, resource)
     {
         m_prev = prev;
--- Tip/wxWindows/src/generic/wizard.cpp	Fri Aug 17 11:52:02 2001
+++ New/wxWindows/src/generic/wizard.cpp	Thu Oct 25 10:55:26 2001
@@ -3,5 +3,8 @@
 // Purpose:     generic implementation of wxWizard class
 // Author:      Vadim Zeitlin
-// Modified by:
+// Modified by: Robert Cavanaugh
+//              1) Added capability for wxWizardPage to accept resources
+//              2) Added "Help" button handler stub
+//              3) Fixed ShowPage() bug on displaying bitmaps
 // Created:     15.08.99
 // RCS-ID:      $Id: wizard.cpp,v 1.19 2001/08/17 17:52:02 VZ Exp $
@@ -74,7 +81,20 @@
 // ----------------------------------------------------------------------------
 
-wxWizardPage::wxWizardPage(wxWizard *parent, const wxBitmap& bitmap)
-            : wxPanel(parent), m_bitmap(bitmap)
+wxWizardPage::wxWizardPage(wxWizard *parent, 
+                            const wxBitmap& bitmap,
+                            char *resource)
+            : wxPanel(parent)
 {
+    // Added 8/20/2001 Robert Cavanaugh
+    // Allow loading controls from dialog editor scripts
+    if(resource != NULL)
+    {
+        if((LoadFromResource(this, resource)) == FALSE)
+        {
+            wxFAIL_MSG(wxT("wxWizardPage LoadFromResource failed!!!!"));
+        }
+    }
+
+    m_PageBitmap = bitmap;
     // initially the page is hidden, it's shown only when it becomes current
     Hide();
--- Tip/wxWindows/src/common/resource.cpp	Sat Aug 18 11:17:14 2001
+++ New/wxWindows/src/common/resource.cpp	Fri Oct 12 13:25:42 2001
@@ -3,5 +3,6 @@
 // Purpose:     Resource system
 // Author:      Julian Smart
-// Modified by:
+// Modified by: Robert Cavanaugh
+//              Allow LoadFromResources() to be used with Dialog Editor scripts
 // Created:     04/01/98
 // RCS-ID:      $Id: resource.cpp,v 1.77 2001/08/18 17:17:13 RR Exp $
@@ -3146,4 +3147,9 @@
     wxString name = resource->GetName();
 
+    // Added 8/21/2001 Robert Cavanaugh
+    // allows windows to accept resources in the
+    // class constructor
+    if(this != parent)
+    {
     if (IsKindOf(CLASSINFO(wxDialog)))
     {
@@ -3167,4 +3173,5 @@
         if (!((wxWindow *)this)->Create(parent, -1, wxPoint(x, y), wxSize(width, height), theWindowStyle, name))
             return FALSE;
+        }
     }


 	  	 
